buildkite/resource_organization_settings_test.go

```diff
@@ -0,0 +1,154 @@
+					// Check that the second IP added to the list is the one we expect, 0.0.0.0/0, this also ensures the length is greater than 1
+					// allowing us to assert the first IP is also added correctly
+					resource.TestCheckResourceAttr("buildkite_organization_settings.let_them_in", "allowed_api_ip_addresses.1", "1.0.0.1/32"),
+				),
+			},
+
+			{
+				Config: testAccOrganizationSettingsConfigBasic([]string{"0.0.0.0/0", "4.4.4.4/32"}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm that the allowed IP addresses are set correctly in Buildkite's system
+					testAccCheckOrganizationSettingsRemoteValues("testkite", []string{"0.0.0.0/0", "4.4.4.4/32"}),
+					// This check allows us to ensure that TF still has access (0.0.0.0/0) and that the new IP address is added correctly
+					resource.TestCheckResourceAttr("buildkite_organization_settings.let_them_in", "allowed_api_ip_addresses.1", "4.4.4.4/32"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccOrganizationSettings_import(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOrganizationSettingsResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccOrganizationSettingsConfigBasic([]string{"0.0.0.0/0", "1.1.1.1/32", "1.0.0.1/32"}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm that the allowed IP addresses are set correctly in Buildkite's system
+					testAccCheckOrganizationSettingsRemoteValues("testkite", []string{"0.0.0.0/0", "1.1.1.1/32", "1.0.0.1/32"}),
+					// Check that the second IP added to the list is the one we expect, 0.0.0.0/0, this also ensures the length is greater than 1
+					// allowing us to assert the first IP is also added correctly
+					resource.TestCheckResourceAttr("buildkite_organization_settings.let_them_in", "allowed_api_ip_addresses.1", "1.0.0.1/32"),
+				),
+			},
+			{
+				ResourceName:      "buildkite_organization_settings.let_them_in",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccOrganizationSettings_disappears(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOrganizationSettingsResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccOrganizationSettingsConfigBasic([]string{"0.0.0.0/0", "1.1.1.1/32", "1.0.0.1/32"}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm that the allowed IP addresses are set correctly in Buildkite's system
+					testAccCheckResourceDisappears(testAccProvider, resourceOrganizationSettings(), "buildkite_organization_settings.let_them_in"),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccOrganizationSettingsConfigBasic(ip_addresses []string) string {
```

Comment:
I think you'll need to iterate over the list to build up a string or maybe json marshal it or something - tests are failing with an incorrect datatype